src/definition/device.js

```diff
@@ -1,40 +1,48 @@
-  ipadPro12Point9Inch1stGeneration: {name: 'iPad Pro (12.9-inch) (1st generation)', display: 'ipad2048x2732d264'},
   ipadPro12Point9Inch2ndGeneration: {name: 'iPad Pro (12.9-inch) (2nd generation)', display: 'ipad2048x2732d264'},
   ipadPro12Point9Inch3rdGeneration: {name: 'iPad Pro (12.9-inch) (3rd generation)', display: 'ipad2048x2732d264'},
+  ipadPro12Point9Inch4thGeneration: {name: 'iPad Pro (12.9-inch) (4th generation)', display: 'ipad2048x2732d264'},
+  ipadPro12Point9Inch: {name: 'iPad Pro (12.9-inch)', display: 'ipad2048x2732d264'},
   ipadPro9Point7Inch: {name: 'iPad Pro (9.7-inch)', display: 'ipad1536x2048d264'},
-  iphone11: {name: 'iPhone 11', display: 'iphone828x1792'},
-  iphone11Pro: {name: 'iPhone 11 Pro', display: 'iphone1125x2436'},
-  iphone11ProMax: {name: 'iPhone 11 Pro Max', display: 'iphone1242x2688'},
-  iphone5: {name: 'iPhone 5', display: 'iphone640x1136'},
-  iphone5c: {name: 'iPhone 5c', display: 'iphone640x1136'},
-  iphone5s: {name: 'iPhone 5s', display: 'iphone640x1136'},
-  iphone6: {name: 'iPhone 6', display: 'iphone750x1334'},
-  iphone6Plus: {name: 'iPhone 6 Plus', display: 'iphone1242x2208'},
-  iphone6s: {name: 'iPhone 6s', display: 'iphone750x1334'},
-  iphone6sPlus: {name: 'iPhone 6s Plus', display: 'iphone1242x2208'},
-  iphone7: {name: 'iPhone 7', display: 'iphone750x1334'},
-  iphone7Plus: {name: 'iPhone 7 Plus', display: 'iphone1242x2208'},
-  iphone8: {name: 'iPhone 8', display: 'iphone750x1334'},
-  iphone8Plus: {name: 'iPhone 8 Plus', display: 'iphone1242x2208'},
-  iphoneSe: {name: 'iPhone SE', display: 'iphone640x1136'},
-  iphoneX: {name: 'iPhone X', display: 'iphone1125x2436'},
-  iphoneXr: {name: 'iPhone XR', display: 'iphone828x1792'},
-  iphoneXs: {name: 'iPhone XS', display: 'iphone1125x2436'},
-  iphoneXsMax: {name: 'iPhone XS Max', display: 'iphone1242x2688'},
-  ipodTouch5thGeneration: {name: 'iPod touch (5th generation)', display: 'iphone640x1136'},
-  ipodTouch6thGeneration: {name: 'iPod touch (6th generation)', display: 'iphone640x1136'},
+  iphone11: {name: 'iPhone 11', display: 'iphone828x1792d326'},
+  iphone11Pro: {name: 'iPhone 11 Pro', display: 'iphone1125x2436d458'},
+  iphone11ProMax: {name: 'iPhone 11 Pro Max', display: 'iphone1242x2688d458'},
+  iphone12: {name: 'iPhone 12', display: 'iphone1170x2532d460'},
+  iphone12Mini: {name: 'iPhone 12 Mini', display: 'iphone1080x2340d476'},
+  iphone12Pro: {name: 'iPhone 12 Pro', display: 'iphone1170x2532d460'},
+  iphone12ProMax: {name: 'iPhone 12 Pro Max', display: 'iphone1284x2778d458'},
+  iphone6s: {name: 'iPhone 6s', display: 'iphone750x1334d326'},
+  iphone6sPlus: {name: 'iPhone 6s Plus', display: 'iphone1242x2208d462'},
+  iphone7: {name: 'iPhone 7', display: 'iphone750x1334d326'},
+  iphone7Plus: {name: 'iPhone 7 Plus', display: 'iphone1242x2208d462'},
+  iphone8: {name: 'iPhone 8', display: 'iphone750x1334d326'},
+  iphone8Plus: {name: 'iPhone 8 Plus', display: 'iphone1242x2208d462'},
```

Comment:
8 Plus is 1920 x 1080 at 401

Reply:
Yeah the "Plus" versions of 6/6s/7/8 all have wacky shit going on. They report a higher resolution to software than the actual physical display is capable of, then they downsample it. The numbers here are the correct, because the higher numbers are what the browser actually "sees". See [iPhone 6 Screens Demystified](https://www.paintcodeapp.com/news/iphone-6-screens-demystified).

I didn't actually change the details of these displays anyway, I just added the density numbers to the end of the display "names". It's probably easier to actually use [this commit](https://github.com/iconduit/iconduit/pull/69/commits/4b7dd0f2d3e830e7e605dd0ea9f84d1e22d50ced) to see the more important changes going on.